include/boost/numeric/ublas/tensor/strides.hpp

```diff
@@ -10,242 +10,103 @@
 //
 /// \file strides.hpp Definition for the basic_strides template class
 
+#ifndef _BOOST_UBLAS_TENSOR_STRIDES_HPP_
+#define _BOOST_UBLAS_TENSOR_STRIDES_HPP_
+
+#include <boost/numeric/ublas/tensor/dynamic_strides.hpp>
+#include <boost/numeric/ublas/tensor/static_strides.hpp>
+
+namespace boost::numeric::ublas {
+
+template <class LStrides, class RStrides,
+  typename std::enable_if_t< detail::is_strides<LStrides>::value && detail::is_strides<RStrides>::value, int > = 0
+>
+bool operator==(LStrides const &lhs,
+                RStrides const &rhs) {
+  if (lhs.size() != rhs.size()) {
+    return false;
+  }
+  for (auto i = 0u; i < lhs.size(); i++) {
+    if (lhs.at(i) != rhs.at(i)) {
+      return false;
+    }
+  }
+  return true;
+}
```

Comment:
I have to use enable_if here because if I don't it will satisfy every type with `operator==` and will give redefinition error. Here we necessarily need concept otherwise it will not work.

Reply:
add the `std::equal` but could not remove `std::enable_if_t` due to unconstrained input

Reply:
ok.